### ISSUE TITLE
Missing props validation when use setPropTypes of recompose

### ISSUE THREAD

**Issue by dumpsayamrat** [yannickcr/eslint-plugin-react#1357]
Hi guys,

I'm on development using react-native recompose and eslint. And I found several error about prop-types validation — `setPropTypes`, `mapProps`, `withHandlers`, etc. 

```js
import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';
import { compose, setPropTypes, onlyUpdateForPropTypes } from 'recompose';

const propTypes = {
  displayText: PropTypes.string.isRequired,
};

const enhance = compose(
  onlyUpdateForPropTypes
  setPropTypes(propTypes),
);

const TheComponent = props => (
  <View>
    <Text>{props.displayText}</Text> // 'displayText' is missing in props validation (react/prop-types)
  </View>
);

export default enhance(TheComponent);
```

Is there any support this situation on next version of `eslint-plugin-react` ?

**Comment by ljharb** [comment]
What's the error? `TheComponent` is an SFC that takes props - you need `TheComponent.propTypes = propTypes;`. Any `setPropTypes` middleware only applies to the wrapped HOC coming out of `enhance` - which is good - but you also need them on the wrapper - on `TheComponent`.

**Comment by dumpsayamrat** [comment]
Oh, You're right. Maybe I'm overthinking.

Thank you.

**Comment by jeggy** [comment]
For a component that would look something like this, would still not work

```jsx
import React from 'react';
import PropTypes from 'prop-types';
import { compose, setPropType } from 'recompose';
import { injectIntl } from 'react-intl';

const propTypes = {
  messageId: PropTypes.string.isRequired,
};

const TheComponent = ({ intl: { formatMessage }, messageId }) => (
  <div>
    <span>{formatMessage(messageId)}</span>
  </div>
);

export default compose(
  injectIntl,
  setPropTypes(propTypes),
)(TheComponent);
```

as the `messageId` needs to be defined inside propTypes, but the `intl` is going to be injected, so it should not be there.

**Comment by ljharb** [comment]
It doesn’t matter that it’s injected; TheComponent should still declare that it needs it.
